dkpro-core-io-nyt-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/nyt/NYTCollectionReader.java

```diff
@@ -0,0 +1,170 @@
+import org.apache.uima.cas.CASException;
+import org.apache.uima.collection.CollectionException;
+import org.apache.uima.fit.component.CasCollectionReader_ImplBase;
+import org.apache.uima.fit.descriptor.ConfigurationParameter;
+import org.apache.uima.jcas.JCas;
+import org.apache.uima.jcas.cas.StringArray;
+import org.apache.uima.resource.ResourceInitializationException;
+import org.apache.uima.util.Progress;
+import org.apache.uima.util.ProgressImpl;
+
+import com.nytlabs.corpus.NYTCorpusDocument;
+
+import de.tudarmstadt.ukp.argumentext.demonstrator.metadata.NYTArticleMetaData;
+import de.tudarmstadt.ukp.dkpro.core.api.metadata.type.DocumentMetaData;
+
+public class NYTCollectionReader extends CasCollectionReader_ImplBase {
+
+	private static final Log LOGGER = LogFactory.getLog(NYTCollectionReader.class);
+
+	/**
+	 * Path to the corpus' data directory.
+	 */
+	public static final String PARAM_DATA_PATH = "dataPath";
+	@ConfigurationParameter(name = PARAM_DATA_PATH, mandatory = true)
+	private String dataPathString;
+
+	/**
+	 * The number of documents which will be skipped at the beginning.
+	 */
+	public static final String PARAM_OFFSET = "offset";
+	@ConfigurationParameter(name = PARAM_OFFSET, mandatory = false)
+	private int offset = 0;
+
+	private int skipped = 0;
+
+	/**
+	 * The total number of documents which will be read.
+	 */
+	public static final String PARAM_LIMIT = "limit";
+	@ConfigurationParameter(name = PARAM_LIMIT, mandatory = false)
+	private int limit = -1;
+
+	private int completed = 0;
+
+	private NYTIterator corpusIterator;
+
+	public void initialize(UimaContext context) throws ResourceInitializationException {
+		super.initialize(context);
+		Path dataPath = Paths.get(this.dataPathString);
+		try {
+			this.corpusIterator = new NYTIterator(dataPath);
+		} catch (IOException | ParserConfigurationException e) {
+			throw new ResourceInitializationException(e);
+		}
+	}
+
+	private DocumentMetaData createDocumentMetaData(JCas aJCas, NYTCorpusDocument doc) {
+		DocumentMetaData metadata = DocumentMetaData.create(aJCas);
+		metadata.setLanguage(java.util.Locale.US.toString());
+		metadata.setDocumentTitle(doc.getHeadline());
+		metadata.setDocumentId(Integer.toString(doc.getGuid()));
+		metadata.setDocumentUri(doc.getSourceFile().getPath());
+		metadata.setCollectionId("NYT_CORPUS");
+		return metadata;
+	}
+
+	@Override
+	public void getNext(CAS aCAS) throws IOException, CollectionException {
+		JCas aJCas;
+		try {
+			aJCas = aCAS.getJCas();
+		} catch (CASException e) {
+			throw new CollectionException();
+		}
+
+		while (isBelowOffset()) {
+			this.corpusIterator.next();
+			skipped++;
+		}
+
+		NYTCorpusDocument doc = this.corpusIterator.next();
+		
+		LOGGER.info("Retrieved " + doc.getSourceFile().toString());
+
+		String body = doc.getBody();
+		if(body != null) {
+			aJCas.setDocumentText(body);
+		} else {
+			aJCas.setDocumentText("");
+		}
+
+		createDocumentMetaData(aJCas, doc);
```

Comment:
This should be done by calling the initCas(aCAS, res) method from JCasCollectionReader_ImplBase.